apps/userdashboard/assets/js/a4_candy_userdashboard/HoverButton.jsx

```diff
@@ -0,0 +1,22 @@
+import React, { useState } from 'react'
+
+export const HoverButton = (props) => {
+  const [buttonText, setButtonText] = useState(props.children.text)
+
+  return (
+    <button
+      id={props.id}
+      className={props.className}
+      type="button"
+      onClick={props.onClick}
+      disabled={props.disabled}
+      onMouseEnter={() => setButtonText(props.children.hoverText)}
+      onMouseLeave={() => setButtonText(props.children.text)}
```

Comment:
is there a reason not to just use a title attribute here?

Reply:
i am not following i think. Do you mean adding attributes for title, hovertext and icon? yes that would be possible, thought its more natural to place the children between the tags.

Reply:
you shouldn't need the mouseEnter or mouseLeave stuff at all unless i'm missing something? https://www.w3schools.com/tags/att_global_title.asp

Reply:
could you please make an example how to use this for the hover effect? i am not following...sorry

Reply:
is the link not working? it's explains the attribute and shows an example. If all we need is a tooltip on hover then we should be able to use that attribute, if we need additional functionality then maybe not? I am unsure as there isn't a US for it?!

Reply:
ahhh now I get it, cheers! ok not sure that was working when I checked, will look at it again